src/sensai/hyperopt.py

```diff
@@ -152,49 +152,32 @@ def _saveCSV(self):
                 os.makedirs(dirname, exist_ok=True)
             self.df.to_csv(self.csvPath, index=False)
 
-    def addModelParamsMetrics(self, model, params, metrics):
-        values = dict(metrics)
-        values.update(params)
-        values["str(model)"] = str(model)
-        self.addValues(values)
-
     def getDataFrame(self) -> pd.DataFrame:
         return self.df
 
 
 class GridSearch(TrackedExperimentDataProvider):
     _log = _log.getChild(__qualname__)
 
-    def __init__(self, modelFactory: Callable[..., VectorModel], parameterOptions: Union[Dict[str, Sequence[Any]], List[Dict[str, Sequence[Any]]]],
+    def __init__(self, modelFactory: Callable[..., VectorModel], parameterOptions: Dict[str, Sequence],
             numProcesses=1, csvResultsPath=None, parameterCombinationSkipDecider: ParameterCombinationSkipDecider = None):
         """
         :param modelFactory: the function to call with keyword arguments reflecting the parameters to try in order to obtain a model instance
-        :param parameterOptions: a dictionary which maps from parameter names to lists of possible values - or a list of such dictionaries,
-            where each dictionary in the list has the same keys
+        :param parameterOptions: a dictionary which maps from parameter names to lists of possible values
```

Comment:
Why would you remove the option to provide a list? This reduces functionality.

Reply:
You stated: "It is probably no too much to ask a user to call update if they have a list of dicts already"
Update will not achieve what the list supports - to the contrary. Note that the list must contain dictionaries where the keys are already all the same, so update would give you the last dictionary only.
The purpose of the list is to avoid full combinatorics. For example, if it only makes sense to explore activation=sigmoid in combination with targetTransformer=maxAbsScaler and activation=none in combination with targetTransformer=standardScaler, then you can have two dictionaries which otherwise contain the same parameter options for any further parameters.

Reply:
I misunderstood the purpose of avoiding full combinatorics, I'll revert it